pubsub/drivertest/drivertest.go

```diff
@@ -245,6 +249,52 @@ func testSendReceive(t *testing.T, newHarness HarnessMaker) {
 	}
 }
 
+// testReceiveSend checks that a receive will still return even if the next
+// message comes in after it was called.
+func testReceiveSend(t *testing.T, newHarness HarnessMaker) {
+	// Set up.
+	ctx := context.Background()
+	h, err := newHarness(ctx, t)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer h.Close()
+	top, sub, cleanup, err := makePair(ctx, h, t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup()
+
+	// Start a receive in the background.
+	gotChan := make(chan *pubsub.Message)
+	go func() {
+		m, err := sub.Receive(ctx)
+		if err != nil {
+			t.Error(err)
+			return
+		}
+		m.Ack()
+		gotChan <- m
+	}()
+
+	// Wait for Receive to block, with any luck.
+	time.Sleep(100 * time.Millisecond)
+
+	// Send a message.
+	want := &pubsub.Message{Body: []byte(fmt.Sprintf("%d", rand.Int()))}
```

Comment:
Why `rand`? Won't that make record/replay problematic?

Reply:
I like to use rand in tests to make it clear that the specific value isn't important, but I see your point. If the tests run in the same order every time then we'll get the same random numbers each time (https://github.com/golang/go/issues/16486), but their order could change or become non-deterministic if we run them in parallel.

Replacing it with something deterministic.